planner/core/exhaust_physical_plans.go

```diff
@@ -1774,17 +1771,28 @@ func (p *LogicalProjection) TryToGetChildProp(prop *property.PhysicalProperty) (
 }
 
 func (p *LogicalProjection) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]PhysicalPlan, bool) {
-	newProp, ok := p.TryToGetChildProp(prop)
-	if !ok {
-		return nil, true
+	allTaskTypes := []property.TaskType{prop.TaskTp}
+	// TODO: only for TiFlash now, support push down projection in TiKV later
+	_, tikv := p.SCtx().GetSessionVars().GetIsolationReadEngines()[kv.TiKV]
```

Comment:
It cannot prove that if the child is using tiflash or tikv. It only means "we have probablity to use tikv". As I have mentioned above/ It's better to pass the property's task type through and check the store type in "attach task"